### PR TITLE
mysql: fix incorrect cast date to int.

### PR DESCRIPTION
**Pull request by overvenus**
see issue #1046 

Thanks @shenli , you guys are awesome!

### CODE REVIEW

File: mysql/time.go [hunk 1]
```diff
@@ -215,17 +215,26 @@ func (t *Time) UnmarshalInLocation(b []byte, loc *time.Location) error {
 }
 
 const numberFormat = "20060102150405"
+const dateFormat = "20060102" // see issue #1046
 
 // ToNumber returns a formatted number.
 // e.g,
 // 2012-12-12T10:10:10 -> 20121212101010
 // 2012-12-12T10:10:10.123456 -> 20121212101010.123456
```
Comment (shenli) [inline mysql/time.go]: Add a new example in the comment to make it consist with the new implementation.

File: mysql/time.go [hunk 2]
```diff
@@ -215,17 +215,26 @@ func (t *Time) UnmarshalInLocation(b []byte, loc *time.Location) error {
...
 func (t Time) ToNumber() Decimal {
+	var tfStr string
```
Comment (shenli) [inline mysql/time.go]: Move this declaration below the if will be better.

**Comment by shenli** [comment]
Thanks! @overvenus
@qiuyesuifeng @coocood PTAL

File: mysql/time.go [hunk 3]
```diff
@@ -215,17 +215,26 @@ func (t *Time) UnmarshalInLocation(b []byte, loc *time.Location) error {
...
+
 	if t.IsZero() {
 		return ZeroDecimal
 	}
 
-	tfStr := numberFormat
+	// See issue #1046
```
Comment (coocood) [inline mysql/time.go]: I think "See issue" is not an appropriate comment in code.

Comment (shenli) [inline mysql/time.go]: Agree.
"Fix issue #1046" is better.

Comment (overvenus) [inline mysql/time.go]: @shenli @coocood ok

**Comment by shenli** [comment]
LGTM
Please squash your commits to a single commit after you get two LGTM.

**Comment by coocood** [comment]
LGTM

**Comment by overvenus** [comment]
@shenli PTAL

cc @c4pt0r

**Comment by shenli** [comment]
Well done!  @overvenus

**Pull request merged**